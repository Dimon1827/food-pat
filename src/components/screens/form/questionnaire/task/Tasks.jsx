import './Task.scss';
import tasksData from './tasksData';
import RadioItem from './RadioItem';

function Tasks() {
  return (
    <fieldset className="questionnaire__field task">
      <legend className="visually-hidden">Основная задача</legend>
      <ul className="task__list line">
        {tasksData.length > 0 &&
          tasksData.map((task, index) => (
            <RadioItem
              key={index}
              type={tasksData.type}
              {...task}
            />
          ))}
      </ul>
    </fieldset>
  );
}

export default Tasks;
