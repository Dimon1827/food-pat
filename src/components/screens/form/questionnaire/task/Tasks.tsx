import './Task.scss';
import tasksData from './tasksData';
import RadioItem from './RadioItem';

function Tasks() {
  return (
    <fieldset className="questionnaire__field task">
      <legend className="visually-hidden">Основная задача</legend>
      <ul className="task__list line">
        {tasksData.length > 0 &&
          tasksData.map(({ id, type, name }) => (
            <RadioItem key={id} id={id} type={type} name={name} />
          ))}
      </ul>
    </fieldset>
  );
}

export default Tasks;
