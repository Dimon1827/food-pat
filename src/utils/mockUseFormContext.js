import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


