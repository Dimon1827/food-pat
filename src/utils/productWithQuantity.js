const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
