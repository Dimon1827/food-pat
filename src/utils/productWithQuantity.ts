import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
