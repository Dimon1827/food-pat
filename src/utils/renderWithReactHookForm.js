import { render } from '@testing-library/react';
import { FormProvider, useForm } from 'react-hook-form';

export function renderWithReactHookForm(ui, { defaultValues = {} } = {}) {
  let reactHookFormMethods = {};

  const Wrapper = ({ children }) => {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}