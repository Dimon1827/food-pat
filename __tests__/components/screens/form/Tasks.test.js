import { screen } from '@testing-library/react';
import { renderWithReactHookForm } from '@/utils/renderWithReactHookForm';
import Tasks from '@/components/screens/form/questionnaire/task/Tasks';

describe('Tasks', () => {
  it('Tasks renders correctly as radio list', () => {
    renderWithReactHookForm(<Tasks />);
    const list = screen.getByRole('group');
    const listItem = screen.getAllByRole('listitem');
    const input = screen.getAllByRole('radio');
    expect(list).toMatchSnapshot();
    expect(listItem).toHaveLength(3);
    expect(input).toHaveLength(3);
  });
});
