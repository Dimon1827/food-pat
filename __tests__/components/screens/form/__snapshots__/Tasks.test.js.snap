// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks Tasks renders correctly as radio list 1`] = `
<fieldset
  class="questionnaire__field task"
>
  <legend
    class="visually-hidden"
  >
    Основная задача
  </legend>
  <ul
    class="task__list line"
  >
    <li
      class="task__item"
    >
      <input
        class="visually-hidden control-input task__input"
        id="slimming"
        name="task"
        type="radio"
        value="slimming"
      />
      <label
        class="label task__label"
        for="slimming"
      >
        Похудение
      </label>
    </li>
    <li
      class="task__item"
    >
      <input
        class="visually-hidden control-input task__input"
        id="gain"
        name="task"
        type="radio"
        value="gain"
      />
      <label
        class="label task__label"
        for="gain"
      >
        Набор массы
      </label>
    </li>
    <li
      class="task__item"
    >
      <input
        class="visually-hidden control-input task__input"
        id="fit"
        name="task"
        type="radio"
        value="fit"
      />
      <label
        class="label task__label"
        for="fit"
      >
        Поддержка формы
      </label>
    </li>
  </ul>
</fieldset>
`;
